libs/@blockprotocol/graph/src/types/temporal-versioning.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Types used in embedding applications and blocks that support multi-axis temporal versioning schemes.
+ */
+
+import { Subtype } from "../util.js";
+
+/**
+ * An ISO 8601 formatted timestamp string
+ */
+export type Timestamp = string;
+
+/**
+ * @todo - doc
+ */
+export type TemporalAxes = "transactionTime" | "decisionTime";
+
+/**
+ * The bound of a time-interval that is either exclusively or inclusively limited by a `Timestamp`
+ */
+export type TimestampLimitedTemporalBound = {
+  kind: "inclusive" | "exclusive";
+  limit: Timestamp;
+};
+
+export type InclusiveTimestampLimitedTemporalBound = Subtype<
```

Comment:
What are we going to use these for?

Reply:
These will end up used in my next PR
<img width="360" alt="image" src="https://user-images.githubusercontent.com/25749103/216044069-865dcc87-1584-435e-9fb0-0a08bf295dff.png">

I wasn't even sure about exporting the intermediary types but it can be a bit annoying when a package doesn't export those types and you're trying to manipulate an object or create a type guard and you can't just reuse their intermediary ones and have to define your own or extract from the type somehow.

Reply:
I'm not really any the wiser from that screenshot but we can talk about it in that PR :)